* 2022-12-14 hub
https://hgwdev.gi.ucsc.edu/~markd/lrgasp/experimental-eval/hub/hub.txt

* WARMING
bamToBed -bed12 does not do the right thing with the U21 reference deletion case

** data/grch38/sequencing
* Angela:
Our ONT amplicon sequencing for validation should be available here:
You will likely need to be on the UCSC VPN to access.
My student Brandon did the basecalling and set up the server. I asked him to put all the FASTQ file pass reads into one file. Here is what he said he did:
"I've added a pass.fastq file into the 221116-basecalled-260bps directory, from running cat pass/*fastq >pass.fastq in order to merge all the passed fastq files"

** data/grch38/sequencing/221116-basecalled-260bps

* get targets without target regions (color 0,128,0)
cd references
tawk '$9=="0,128,0"'  ../../../hub/hg38/targets.bed >target-trans.bed

* ALIGNMENT
** align ONT reads to genome
cd align
(time nice minimap2 -t 64 -ax splice --junc-bed=../reference/target-trans.bed /hive/users/markd/gencode/projs/lrgasp/experimental-eval/data/grch38/reference/lrgasp_grch38_sirvs.fasta.gz   ../sequencing/221116-basecalled-260bps/runid_7f2d5c3b79ee43d49f5db857aca2eaa618ab7879.fastq.gz  | samtools sort -O BAM > ont-rna-wtc11-genome.bam)>&ont.log&

** align PB reads to genome
cd align
(time nice minimap2 -t 64 -ax splice:hq --junc-bed=../reference/target-trans.bed /hive/users/markd/gencode/projs/lrgasp/experimental-eval/data/grch38/reference/lrgasp_grch38_sirvs.fasta.gz  ../../pb-sequencing-mixed/st183-lrgasp/ST183-LRGASP.hifi_reads.fastq.gz  | samtools sort -O BAM > pb-rna-wtc11-genome.bam)>&pb.log&

** control alignment of amplicons to genome
(time nice minimap2 -t 64 -ax splice:hq --junc-bed=../reference/target-trans.bed /hive/users/markd/gencode/projs/lrgasp/experimental-eval/data/grch38/reference/lrgasp_grch38_sirvs.fasta.gz   ../../grch38-transcriptome/reference/wtc11-amplicons.fa  | samtools sort -O BAM > cntl-rna-wtc11-genome.bam)>&cntl.log&

for f in *.bam ; do samtools index $f & done
ln -f *.bam* ../../../hub/hg38/

** amplicons target
cd grch38-transcriptome/reference
tawk '$3==1{print ">" $1 " " $2;print $9}' ../../../../primers/primer-design/hub/hg38/juju_designs.hg38.isoforms.tsv  >wtc11-amplicons.fa
samtools faidx wtc11-amplicons.fa 
ln wtc11-amplicons.fa* ../../../hub/wtc11-amplicons/

** ONT amplcons alignments
cd data/grch38-transcriptome/align
(time nice minimap2 -ax map-ont -t 32 ../reference/wtc11-amplicons.fa ../sequencing/wtc11.ont.fastq | samtools sort -O BAM >ont-rna-wtc11-amplicons.bam)>&1.log&
nice samtools index -@ 32 ont-rna-wtc11-amplicons.bam 

(time nice minimap2 -ax map-pb -t 32 ../reference/wtc11-amplicons.fa ../../pb-sequencing-mixed/st183-lrgasp/ST183-LRGASP.hifi_reads.fastq.gz  | samtools sort -O BAM > pb-rna-wtc11-amplicons.bam)>&2.log&
nice samtools index -@ 32 pb-rna-wtc11-amplicons.bam


** control alignment of amplicons to amplicons
cd grch38-transcriptome/align/
(time nice minimap2 -t 64 -ax map-pb ../reference/wtc11-amplicons.fa   ../../grch38-transcriptome/reference/wtc11-amplicons.fa  | samtools sort -O BAM > cntl-rna-wtc11-amplicons.bam)>&log&



* BROWSER
cd hub/hg38
../../bin/mkBamsTrackDb *rna-wtc11*.bam >filtered.trackDb.txt
bedToHtmlDir -public-browser -sort-by-name -title "LRGASP experimental validations targets" -hub https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hub.txt -hub http://conesalab.org/LRGASP/LRGASP_hub/hub.txt hg38 targets.bed dir

* ANALYZE
** intron chain with indel=2 filter
cd data/grch38/analyze
for prot in pb ont cntl ; do
   ../../../bin/genomeAlignAnalyze --maxIndelBases=2 ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/amplicons.bed ../align/${prot}-rna-wtc11-genome.bam --supportBam=${prot}-rna-wtc11.intron-chains.indel=2.bam ${prot}-rna-wtc11.intron-chains.indel=2.tsv  &
done >&2.log

for prot in pb ont cntl ; do
   samtools index ${prot}-rna-wtc11.intron-chains.indel=2.bam &
done
ln -f *.bam* ../../../hub/hg38/

** intron chain ( no indel filter)
cd data/grch38/analyze
for prot in pb ont cntl ; do
   ../../../bin/genomeAlignAnalyze ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/amplicons.bed ../align/${prot}-rna-wtc11-genome.bam --supportBam=${prot}-rna-wtc11.intron-chains.indel=any.bam ${prot}-rna-wtc11.intron-chains.indel=any.tsv  &
done >&log

for prot in pb ont cntl ; do
   samtools index ${prot}-rna-wtc11.intron-chains.indel=any.bam &
done
ln -f *.bam* ../../../hub/hg38/

** data/grch38-transcriptome/grch38-transcriptome/analyze.sim
for sim in 1.0 0.99 0.98 0.95 0.90 ; do for plat in pb ont cntl ; do
   nice ../bin/simAmpSupport ${plat} ${sim}&
done ; done >&log   
ln -f *.genome.bam*  ../../../hub/hg38/
ln -f *.amplicon.bam*  ../../../hub/wtc11-amplicons/

** data/grch38-transcriptome/analyze.indel
for indel in 0 1 2; do for plat in pb ont cntl ; do
   nice ../bin/indelAmpSupport ${plat} ${indel}&
done; done >&log   
ln -f *.genome.bam*  ../../../hub/hg38/
ln -f *.amplicon.bam*  ../../../hub/wtc11-amplicons/

* SUMMARIES
** cd grch38-transcriptome/analyze.sim
../../../bin/summarizeSupport similarity ../../../hub/hg38/juju_designs.hg38.isoforms.tsv --genomeStatsTsv=../../grch38/analyze/ont-rna-wtc11.intron-chains.indel=any.tsv --genomeStatsTsv=../../grch38/analyze/pb-rna-wtc11.intron-chains.indel=any.tsv   
--genomeStatsTsv=../../grch38/analyze/cntl-rna-wtc11.intron-chains.indel=any.tsv *-wtc11.sim=*.tsv >wtc11-support.sim.tsv
 cp wtc11-support.sim.tsv ../../../analyze/
** cd grch38-transcriptome/analyze.indel
../../../bin/summarizeSupport indels  ../../../hub/hg38/juju_designs.hg38.isoforms.tsv  --genomeStatsTsv=../../grch38/analyze/ont-rna-wtc11.intron-chains.indel=2.tsv --genomeStatsTsv=../../grch38/analyze/pb-rna-wtc11.intron-chains.indel=2.tsv  --genomeStatsTsv=../../grch38/analyze/cntl-rna-wtc11.intron-chains.indel=2.tsv  *-wtc11.indel=*.tsv >wtc11-support.indel.tsv
 cp wtc11-support.indel.tsv ../../../analyze/




* spreadsheet coloring
custom format formulas:
=and(mod(row()-2,8) <= 3,$F1:K1=0)
=and(mod(row()-2,8) > 3,$F1:K1=0)
=mod(row()-2,8) <= 3

