* 2022-12-14 hub
https://hgwdev.gi.ucsc.edu/~markd/lrgasp/experimental-eval/hub/hub.txt

* WARMING
bamToBed -bed12 does not do the right thing with the U21 reference deletion case

** data/grch38/sequencing
* Angela:
Our ONT amplicon sequencing for validation should be available here:
You will likely need to be on the UCSC VPN to access.
My student Brandon did the basecalling and set up the server. I asked him to put all the FASTQ file pass reads into one file. Here is what he said he did:
"I've added a pass.fastq file into the 221116-basecalled-260bps directory, from running cat pass/*fastq >pass.fastq in order to merge all the passed fastq files"

* data/grch38/sequencing/221116-basecalled-260bps

* align reads to genome
cd align
minimap2 -ax splice
#  map-ont is the default

find ../sequencing/221116-basecalled-260bps/pass/ -name '*.fastq'  |awk '{print "../bin/runMinimapOnt", $0, "{check out exists tmp/" gensub("^.*/", "", 1) ".sam}"}'  >jobs.para

ssh to ku
# must use 32gb of memory or get empty output

para freeBatch -batch=b1
para create -ram=32g -batch=b1 jobs.para
para try -batch=b1

y# combine into BAM
 
nice samtools merge --output-fmt BAM --threads 32 -c -p  ont-rna-wtc11-genome.bam tmp/*.sam

# check id unqiueness
samtools fasta ont-rna-wtc11-genome.bam | faSize -detailed stdin | cut -f 1 sort | twc -l

total 4880899
uniq  4880899

# didn't really need to split it up.



* align PB reads to genome
cd align
(time nice minimap2 -t 64 -ax splice:hq /hive/users/markd/gencode/projs/lrgasp/experimental-eval/data/grch38/reference/lrgasp_grch38_sirvs.fasta.gz  ../../pb-sequencing-mixed/st183-lrgasp/ST183-LRGASP.hifi_reads.fastq.gz  | samtools sort -O BAM > pb-rna-wtc11-genome.bam)>&log&

nice samtools merge -@ 64 -O BAM -o pb+ont-rna-wtc11-genome.bam pb-rna-wtc11-genome.bam ont-rna-wtc11-genome.bam >&log&

* control alignment of amplicons to genome
(time nice minimap2 -t 64 -ax splice:hq /hive/users/markd/gencode/projs/lrgasp/experimental-eval/data/grch38/reference/lrgasp_grch38_sirvs.fasta.gz   ../../grch38-transcriptome/reference/wtc11-amplicons.fa  | samtools sort -O BAM > cntl-rna-wtc11-genome.bam)>&cntl.log&

for f in *.bam ; do samtools index $f & done
ln -f *.bam* ../../../hub/hg38/

* generate transcriptome of targeted loci
cd grch38-transcriptome/reference

* amplicons target
cd reference
tawk '$3==1{print ">" $1 " " $2;print $9}' ../../../../primers/primer-design/hub/hg38/juju_designs.hg38.isoforms.tsv  >wtc11-amplicons.fa
samtools faidx wtc11-targets.fa

** ONT alignments
cd align
(time nice minimap2 -ax map-ont -t 32 ../reference/wtc11-amplicons.fa ../sequencing/wtc11.ont.fastq | samtools sort -O BAM >ont-rna-wtc11-amplicons.bam)>&log&
real	11m28.550s
nice samtools index -@ 32 ont-rna-wtc11-amplicons.bam 
** PB align
(time nice minimap2 -ax map-hifi -t 32 ../reference/wtc11-amplicons.fa ../../pb-sequencing-mixed/st183-lrgasp/ST183-LRGASP.hifi_reads.fastq.gz  | samtools sort -O BAM > pb-rna-wtc11-amplicons.bam)>&log&
nice samtools index -@ 32 pb-rna-wtc11-amplicons.bam 

** combine
samtools merge -O BAM -o pb+ont-rna-wtc11-amplicons.bam ont-rna-wtc11-amplicons.bam pb-rna-wtc11-amplicons.bam &
nice samtools index -@ 32 pb+ont-rna-wtc11-amplicons.bam 

* control alignment of amplicons to amplicons
cd control/grch38-transcriptome/
(time nice minimap2 -t 64 -ax splice ../reference/wtc11-amplicons.fa   ../../grch38-transcriptome/reference/wtc11-amplicons.fa  | samtools sort -O BAM > cntl-rna-wtc11-amplicons.bam)>&log&


* intron chain with indel=2 filter
cd data/grch38/analyze
for prot in pb ont pb+ont cntl ; do
   ../../../bin/genomeAlignAnalyze --maxIndelBases=2 ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/targets.bed ../align/${prot}-rna-wtc11-genome.bam --supportBam=${prot}-rna-wtc11.intron-chains.indel=2.bam ${prot}-rna-wtc11.intron-chains.indel=2.tsv  &
done >&2.log

for prot in pb ont pb+ont cntl ; do
   samtools index ${prot}-rna-wtc11.intron-chains.indel=2.bam &
done
ln -f *.bam* ../../../hub/hg38/

* intron chain ( no indel filter)
cd data/grch38/analyze
for prot in pb ont pb+ont cntl ; do
   ../../../bin/genomeAlignAnalyze ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/targets.bed ../align/${prot}-rna-wtc11-genome.bam --supportBam=${prot}-rna-wtc11.intron-chains.indel=any.bam ${prot}-rna-wtc11.intron-chains.indel=any.tsv  &
done >&log

for prot in pb ont pb+ont cntl ; do
   samtools index ${prot}-rna-wtc11.intron-chains.indel=any.bam &
done
ln -f *.bam* ../../../hub/hg38/

* data/grch38-transcriptome/grch38-transcriptome/analyze.sim
for sim in 1.0 0.99 0.98 0.95 0.90 ; do for plat in pb ont pb+ont cntl ; do
   nice ../bin/simAmpSupport ${plat} ${sim}&
done ; done >&log   

* data/grch38-transcriptome/analyze.indel
for indel in 0 1 2; do for plat in pb ont pb+ont cntl ; do
   nice ../bin/indelAmpSupport ${plat} ${indel}&
done; done >&log   

* browser
cd hub/hg38
../../bin/mkBamsTrackDb *rna-wtc11*.bam >filtered.trackDb.txt
bedToHtmlDir -public-browser -sort-by-name -title "LRGASP experimental validations targets" -hub https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hub.txt -hub http://conesalab.org/LRGASP/LRGASP_hub/hub.txt hg38 targets.bed dir

* summaries

** cd grch38-transcriptome/analyze.indel
../../../bin/summarizeSupport indels ../../../hub/hg38/juju_designs.hg38.isoforms.tsv --genomeStatsTsv=../../grch38/analyze/ont-rna-wtc11.intron-chains.indel=2.tsv --genomeStatsTsv=../../grch38/analyze/pb-rna-wtc11.intron-chains.indel=2.tsv  --genomeStatsTsv=../../grch38/analyze/pb+ont-rna-wtc11.intron-chains.indel=2.tsv   --genomeStatsTsv=../../grch38/analyze/cntl-rna-wtc11.intron-chains.indel=2.tsv *-wtc11.indel=*.tsv >wtc11-support.indel.tsv
ln -f *.bam* ../../../hub/hg38/
 cp wtc11-support.indel.tsv  ../../../analyze/

** cd grch38-transcriptome/analyze.sim
../../../bin/summarizeSupport similarity ../../../hub/hg38/juju_designs.hg38.isoforms.tsv --genomeStatsTsv=../../grch38/analyze/ont-rna-wtc11.intron-chains.indel=any.tsv --genomeStatsTsv=../../grch38/analyze/pb-rna-wtc11.intron-chains.indel=any.tsv  --genomeStatsTsv=../../grch38/analyze/pb+ont-rna-wtc11.intron-chains.indel=any.tsv   --genomeStatsTsv=../../grch38/analyze/cntl-rna-wtc11.intron-chains.indel=any.tsv *-wtc11.sim=*.tsv >wtc11-support.sim.tsv
ln -f *.bam* ../../../hub/hg38/

 cp wtc11-support.sim.tsv ~/public_html/lrgasp/experimental-eval/
 cp wtc11-support.sim.tsv ../../../analyze/


* experiment with different alignments on controls.
** data/grch38/cntl-expr1.tmp  genome alignments

*** minimap2 splice;hg
(nice minimap2 -t 64 -ax splice:hq  /hive/users/markd/gencode/projs/lrgasp/experimental-eval/data/grch38/reference/lrgasp_grch38_sirvs.fasta.gz ../../grch38-transcriptome/reference/wtc11-amplicons.fa  | samtools sort -O BAM > cntl-mm-splicehq.bam) >&1.log&
samtools index cntl-mm-splicehq.bam 
../../../bin/genomeAlignAnalyze ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/targets.bed cntl-mm-splicehq.bam --supportBam=cntl-mm-splicehq.support.bam cntl-mm-splicehq.support.tsv

tmlr join -j target_id,transcript_id --rp sphq_ -f ../analyze/cntl-rna-wtc11.intron-chains.indel=2.tsv cntl-mm-splicehq.support.tsv >cntl-mm-cmp.tsv
tmlr filter '$chain_supports !=$sphq_chain_supports' cntl-mm-cmp.tsv > cntl-mm-cmp.diff.tsv

# only adds IGF2BP3+1

*** blat
blat -fine -ooc=/hive/data/genomes/hg38/jkStuff/hg38.11.ooc -q=rna  ../reference/lrgasp_grch38_sirvs.fasta.gz ../../grch38-transcriptome/reference/wtc11-amplicons.fa cntl-blat.raw.psl &
pslCDnaFilter -maxAligns=1 cntl-blat.raw.psl  cntl-blat.filt1.psl
  cntl-blat.filt1.psl	168
# psl2sam.pl doesn't flag introns, so tools don't work; just look at stats

faSize -detailed ../../grch38-transcriptome/reference/wtc11-amplicons.fa >wtc11-amplicons.sizes
pslStats -tsv -queryStats -queries=wtc11-amplicons.sizes cntl-blat.filt1.psl cntl-blat.filt1.qstats

# pull out the imprefect ones
tmlr filter '$minIdent < 1.0 ||  $minQCover < 1.0' cntl-blat.filt1.qstats  >cntl-blat.filt1.imperfect.qstats
pslSelect -queries=<(cut -f 1 cntl-blat.filt1.imperfect.qstats) cntl-blat.filt1.psl  cntl-blat.filt1.imperfect.psl 
    cntl-blat.filt1.imperfect.psl	21
pslFmt cntl-blat.filt1.imperfect.psl  > cntl-blat.filt1.imperfect.txt

# imperfect aligments are a small number of bases unaligned at the start/end


* spreadsheet coloring
oocustom format formula: =mod(row()-2,4)=0
