* hub dir https://hgwdev.gi.ucsc.edu/~markd/lrgasp/experimental-eval/hub/hg38/dir/
* IGV tracks
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/targets.bb
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.98.genome.bam
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.95.genome.bam
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.90.genome.bam

* Gloria spread sheet
https://docs.google.com/spreadsheets/d/1nW8W_xtfvKE1sSDBNRsZ3K6HWEtxb4jWwZtPh5hNMVA/edit#gid=0

* JuJu spread sheet
https://docs.google.com/spreadsheets/d/1Y3KhZoezR8Ti5mauYXQFxuwE04xmEgPPyxoJoRxuYuA/edit#gid=210377221

* issues:
** how to get squanti stats for gencode models ENST & TM_

* debugging:
 ../../../bin/genomeAlignAnalyze --target=DUSP4+C ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/targets.bed  ../align/pb-rna-wtc11-genome.bam debug.tsv --supportBam=debug.bam &

evalsDf = pd.read_table("wtc11-support.summary.tsv")
 
* stats for paper
../bin/paperEvalStats wtc11-support.summary.tsv /dev/stdout
GENCODE-known, N=XX:
GENCODE-novel, N=XX:,
GENCODE-suspect, N=XX:
GENCODE-known validation rate, XX%
GENCODE-known that failed to validate n+XX
GENCODE-novel   validation rate, XX %
GENCODE-suspect validation rate of XX,
GENCODE-suspect “validated” XX

novel isoforms count:
novel isoforms detected in N pipelines (different customs)
novel isoforms in less name N pipelines

validation rate for isoforms unique 
we find XX validation rate for isoforms that were not reproducible across pipelines:

In order to maintain a fair comparison, we limited selection of targets to <XX kb,
targets with at least an average TPM of XX

An isoform, X, which was detected in XX of XX ONT pipelines but only XX of XX PacBio pipelines

including cases of a single long read (Figure XX).

We found that [Mark add an example here] deemed validated revealed that an isoform for gene XX had an
incorrect alignment.


