* 2022-12-14 hub
https://hgwdev.gi.ucsc.edu/~markd/lrgasp/experimental-eval/hub/hub.txt

* WARMING
bamToBed -bed12 does not do the right thing with the U21 reference deletion case

** data/grch38/sequencing
* Angela:
Our ONT amplicon sequencing for validation should be available here:
You will likely need to be on the UCSC VPN to access.
My student Brandon did the basecalling and set up the server. I asked him to put all the FASTQ file pass reads into one file. Here is what he said he did:
"I've added a pass.fastq file into the 221116-basecalled-260bps directory, from running cat pass/*fastq >pass.fastq in order to merge all the passed fastq files"

* data/grch38/sequencing/221116-basecalled-260bps

* align reads to genome
cd align
minimap2 -ax splice
#  map-ont is the default

find ../sequencing/221116-basecalled-260bps/pass/ -name '*.fastq'  |awk '{print "../bin/runMinimapOnt", $0, "{check out exists tmp/" gensub("^.*/", "", 1) ".sam}"}'  >jobs.para

ssh to ku
# must use 32gb of memory or get empty output

para freeBatch -batch=b1
para create -ram=32g -batch=b1 jobs.para
para try -batch=b1

# combine into BAM
 
nice samtools merge --output-fmt BAM --threads 32 -c -p  ont-rna-wtc11-genome.bam tmp/*.sam

# check id unqiueness
samtools fasta ont-rna-wtc11-genome.bam | faSize -detailed stdin | cut -f 1 sort | twc -l

total 4880899
uniq  4880899

# didn't really need to split it up.



* align PB reads to genome
cd align
(time nice minimap2 -t 64 -ax splice -p map-hifi /hive/users/markd/gencode/projs/lrgasp/experimental-eval/data/grch38/reference/lrgasp_grch38_sirvs.fasta.gz  ../../pb-sequencing-mixed/st183-lrgasp/ST183-LRGASP.hifi_reads.fastq.gz  | samtools sort -O BAM > pb-rna-wtc11-genome.bam)>&log&

nice samtools merge -@ 64 -O BAM -o pb+ont-rna-wtc11-genome.bam pb-rna-wtc11-genome.bam ont-rna-wtc11-genome.bam >&log&

* control alignment of amplicons to genome
(time nice minimap2 -t 64 -ax splice -p map-hifi /hive/users/markd/gencode/projs/lrgasp/experimental-eval/data/grch38/reference/lrgasp_grch38_sirvs.fasta.gz   ../../grch38-transcriptome/reference/wtc11-amplicons.fa  | samtools sort -O BAM > cntl-rna-wtc11-genome.bam)>&cntl.log&

* generate transcriptome of targeted loci
cd grch38-transcriptome/reference

* amplicons target
cd reference
tawk '$3==1{print ">" $1 " " $2;print $9}' ../../../../primers/primer-design/hub/hg38/juju_designs.hg38.isoforms.tsv  >wtc11-amplicons.fa
samtools faidx wtc11-targets.fa

** ONT alignments
cd align
(time nice minimap2 -ax map-ont -t 32 ../reference/wtc11-amplicons.fa ../sequencing/wtc11.ont.fastq | samtools sort -O BAM >ont-rna-wtc11-amplicons.bam)>&log&
real	11m28.550s
nice samtools index -@ 32 ont-rna-wtc11-amplicons.bam 
** PB align
(time nice minimap2 -ax map-hifi -t 32 ../reference/wtc11-amplicons.fa ../../pb-sequencing-mixed/st183-lrgasp/ST183-LRGASP.hifi_reads.fastq.gz  | samtools sort -O BAM > pb-rna-wtc11-amplicons.bam)>&log&
nice samtools index -@ 32 pb-rna-wtc11-amplicons.bam 

** combine
samtools merge -O BAM -o pb+ont-rna-wtc11-amplicons.bam ont-rna-wtc11-amplicons.bam pb-rna-wtc11-amplicons.bam &
nice samtools index -@ 32 pb+ont-rna-wtc11-amplicons.bam 

* control alignment of amplicons to amplicons
cd control/grch38-transcriptome/
(time nice minimap2 -t 64 -ax splice -p map-hifi ../reference/wtc11-amplicons.fa   ../../grch38-transcriptome/reference/wtc11-amplicons.fa  | samtools sort -O BAM > cntl-rna-wtc11-amplicons.bam)>&log&



* debugging
../../../bin/transcriptomeAlignAnalyze --target=BTG2+B --minSimilarity=0.1 --readStatsTsv=debug.sim=0.1.rstats.tsv --filteredBam=debug.sim=0.1.bam ../align/ont-rna-wtc11-amplicons.bam debug.sim=0.1.tsv &
* analysis
cd analyze
for sim in 1.0 0.99 0.98 0.95 0.90 ; do for plat in pb ont pb+ont cntl ; do
   nice ../bin/simAmpSupport ${plat} ${sim}&
done ; done >&log   

../../../bin/transcriptomeSummary cntl-wtc11.sim=*.tsv >ont-summary.tsv &
../../../bin/transcriptomeSummary ont-rna-wtc11.sim=*.tsv >ont-summary.tsv &
../../../bin/transcriptomeSummary pb-rna-wtc11.sim=*.tsv >pb-summary.tsv &
../../../bin/transcriptomeSummary pb+ont-rna-wtc11.sim=*.tsv >pb+ont-summary.tsv &


../../../bin/transcriptomeFilterCmp --first ont-rna-wtc11.sim=*.tsv >ont.cmp.tsv &
../../../bin/transcriptomeFilterCmp --first pb-rna-wtc11.sim=*.tsv >pb.cmp.tsv &
../../../bin/transcriptomeFilterCmp --first pb+ont-rna-wtc11.sim=*.tsv >pb+ont.cmp.tsv &

ln -f *.bam* ../../../hub/hg38/

* browser
cd hub/hg38
../../bin/mkBamsTrackDb *rna-wtc11*.bam >filtered.trackDb.txt
bedToHtmlDir -public-browser -sort-by-name -title "LRGASP experimental validations targets" -hub https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hub.txt -hub http://conesalab.org/LRGASP/LRGASP_hub/hub.txt hg38 targets.bed dir


* manual review

cut -f 1-5 pb+ont.cmp.tsv | tawk '{st=($3==1.0) ? "perfect" : "??"; print $0, st, ""}' > ../../../doc/wtc11-pd+ont-manual-review.tsv

# to find reads
../../../bin/transcriptomeAlignAnalyze --target=DGAT2+D --minSimilarity=0.99 --readStatsTsv=debug.rstats.tsv ../align/pb+ont-rna-wtc11-amplicons.bam debug.tsv


* check intron chains
cd data/grch38/analyze
for prot in pb ont pb+ont cntl ; do
../../../bin/genomeAlignAnalyze ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/targets.bed ../align/${prot}-rna-wtc11-genome.bam --supportBam=${prot}.tmp.bam ${prot}-rna-wtc11.intron-chains.tsv  &
done >&log

for prot in pb ont pb+ont cntl ; do
   samtools sort ${prot}.tmp.bam > ${prot}-rna-wtc11.intron-chains.bam &
done
rm *.tmp.bam
for prot in pb ont pb+ont cntl ; do
   samtools index  ${prot}-rna-wtc11.intron-chains.bam &
 done
ln -f *.bam* ../../../hub/hg38/

tar -zcf ~/public_html/lrgasp/experimental-eval/wtc11-intron-chains.tar.gz grch38/analyze/

* final summary
cd grch38-transcriptome/analyze

 ../../../bin/summarizeSupport ../../../hub/hg38/juju_designs.hg38.isoforms.tsv --genomeStatsTsv=../../grch38/analyze/ont-rna-wtc11.intron-chains.tsv --genomeStatsTsv=../../grch38/analyze/pb-rna-wtc11.intron-chains.tsv  --genomeStatsTsv=../../grch38/analyze/pb+ont-rna-wtc11.intron-chains.tsv   --genomeStatsTsv=../../grch38/analyze/cntl-rna-wtc11.intron-chains.tsv *-wtc11.sim=*.tsv >wtc11-support.tsv

cp wtc11-support.tsv ~/public_html/lrgasp/experimental-eval/

* debug
../../../bin/genomeAlignAnalyze --target=ALG6+1 ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/targets.bed ../align/pb-rna-wtc11-genome.bed.gz debug.tsv &
