* hub dir https://hgwdev.gi.ucsc.edu/~markd/lrgasp/experimental-eval/hub/hg38/dir/
* IGV tracks
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/targets.bb
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.98.genome.bam
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.95.genome.bam
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.90.genome.bam

* Gloria spread sheet
https://docs.google.com/spreadsheets/d/1nW8W_xtfvKE1sSDBNRsZ3K6HWEtxb4jWwZtPh5hNMVA/edit#gid=0

* JuJu spread sheet
https://docs.google.com/spreadsheets/d/1Y3KhZoezR8Ti5mauYXQFxuwE04xmEgPPyxoJoRxuYuA/edit#gid=210377221


* ALG6+1
All models are bad. Intron 3' end at chr1:63,411,128-63,411,194 has GT-AT
splice site.  This is a documented U12 intron, but with a very rare splicing
pattern.  Supported by WTC-11 reads. GENCODE and RefSeq.  Models and minimap2
seemed to have force GT-AG 6 bases upstream by calling an intron +
target-insertion (D).

* ARPC2+1 (287pb
Lots of 100% amplicon support, but no chain support.
Exon at chr2:218,217,459-218,217,545 is include in all chains, but not in target.
How is it possible


* debugging:
 ../../../bin/genomeAlignAnalyze --target=DUSP4+C ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/targets.bed  ../align/pb-rna-wtc11-genome.bam debug.tsv --supportBam=debug.bam &


* stats for paper
GENCODE-known, N=XX:
GENCODE-novel, N=XX:,
GENCODE-suspect, N=XX:
GENCODE-known  validation rate, XX%
GENCODE-known that failed to validate n+XX
GENCODE-novel   validation rate, XX %
GENCODE-suspect validation rate of XX,
GENCODE-suspect “validated” XX


novel isoforms were detected in this study, XX novel isoforms in WTC11, Fig. 1b).

Some isoforms were detected consistently (found in more than XX of the XX pipelines) but a vast majority
of isoforms were detected in less than XX pipelines.

we find XX validation rate for isoforms that were not reproducible across pipelines:

In order to maintain a fair comparison, we limited selection of targets to <XX kb,

targets with at least an average TPM of XX

An isoform, X, which was detected in XX of XX ONT pipelines but only XX of XX PacBio pipelines
was characterized by being low
abundance.


 including cases of a single long read (Figure XX).

We found that [Mark add an example here] deemed validated revealed that an isoform for gene XX had an
incorrect alignment.

