* hub dir https://hgwdev.gi.ucsc.edu/~markd/lrgasp/experimental-eval/hub/hg38/dir/
* IGV tracks
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/targets.bb
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.98.genome.bam
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.95.genome.bam
https://hgwdev.gi.ucsc.edu/~markd/gencode/lrgasp/experimental-eval/hub/hg38/pb+ont-rna-wtc11.sim=0.90.genome.bam

* Gloria spread sheet
https://docs.google.com/spreadsheets/d/1nW8W_xtfvKE1sSDBNRsZ3K6HWEtxb4jWwZtPh5hNMVA/edit#gid=0

* JuJu spread sheet
https://docs.google.com/spreadsheets/d/1Y3KhZoezR8Ti5mauYXQFxuwE04xmEgPPyxoJoRxuYuA/edit#gid=210377221


* wtc11:  generate amplicon and transcript lengths to add to spread sheet
cd extra-data/wtc11

# wtc11-trans_len.tsv
bigBedToBed ../../../primers/primer-design/data/hg38/WTC11_consolidated.bigBed stdout | cut -f 1-12 >WTC11_consolidated.bed
(echo "transcript${tab}trans_len" ; (twoBitToFa -bed=WTC11_consolidated.bed  /hive/data/genomes/hg38/hg38.2bit  stdout | faSize -detailed /dev/stdin)) >wtc11-trans_len.tsv

# wtc11.isoforms.tsv
tmlr filter '$pri==1' ../../../primers/primer-design/hub/hg38/juju_designs.hg38.isoforms.tsv | tmlr join -j transcript_id -l transcript -r transcript_id -f wtc11-trans_len.tsv | tmlr rename target_id,target,transcript_id,transcript > wtc11.isoforms.tsv 

# wtc11-lengths.rows.tsv
../../bin/spreadSheetOrder --rowsToAddPerGroup=2 wtc11-support.tsv wtc11.isoforms.tsv /dev/stdout | tmlr cut -o -f target,transcript,trans_len,amplicon_len >wtc11-lengths.rows.tsv

* gencode: generate amplicon and transcript lengths to add to spread sheet

# gencode-trans_len.tsv
for bb in  ../../../primers/primer-design/data/hg38/human_GENCODE_tmerge_transcripts.bb ../../../primers/primer-design/data/hg38/non_redundant_*.bb ; do bigBedToBed $bb stdout | cut -f 1-12 ; done > gencode-combined.bed 
(echo "transcript${tab}trans_len" ; (twoBitToFa -bed=gencode-combined.bed /hive/data/genomes/hg38/hg38.2bit stdout) | faSize -detailed stdin) >gencode-trans_len.tsv

# gencode.isoforms.tsv
tmlr filter '$pri==1' ../../../primers/primer-design/hub/hg38/juju_designs.hg38.isoforms.tsv | tmlr join -j transcript_id -l transcript -r transcript_id -f gencode-trans_len.tsv | tmlr rename target_id,target,transcript_id,transcript > gencode.isoforms.tsv 

# gencode-lengths.rows.tsv
../../bin/spreadSheetOrder --rowsToAddPerGroup=2 gencode-support.tsv gencode.isoforms.tsv /dev/stdout | tmlr cut -o -f target,transcript,trans_len,amplicon_len >gencode-lengths.rows.tsv


* match GENCODE models to WTC11 consolidated tracks
gtfToGenePred ../gencode/human_LRGASP_unannotated_transcripts.gtf gencode.gp
genePredToBed gencode.gp gencode.bed
../bin/matchByIntronChain LRGASP_manual_annotation.human.all.20220428.bed WTC11_consolidated-gencode.tsv gencode.unmatched.bed >&out
# nothing useful found
* debugging:
 ../../../bin/genomeAlignAnalyze --target=DUSP4+C ../../../hub/hg38/juju_designs.hg38.isoforms.tsv ../../../hub/hg38/targets.bed  ../align/pb-rna-wtc11-genome.bam debug.tsv --supportBam=debug.bam &

evalsDf = pd.read_table("wtc11-support.summary.tsv")
 
* stats for paper
../bin/paperEvalStats wtc11-support.summary.tsv /dev/stdout
GENCODE-known, N=XX:
GENCODE-novel, N=XX:,
GENCODE-suspect, N=XX:
GENCODE-known validation rate, XX%
GENCODE-known that failed to validate n+XX
GENCODE-novel   validation rate, XX %
GENCODE-suspect validation rate of XX,
GENCODE-suspect “validated” XX

novel isoforms count:
novel isoforms detected in N pipelines (different customs)
novel isoforms in less name N pipelines

validation rate for isoforms unique 
we find XX validation rate for isoforms that were not reproducible across pipelines:

In order to maintain a fair comparison, we limited selection of targets to <XX kb,
targets with at least an average TPM of XX

An isoform, X, which was detected in XX of XX ONT pipelines but only XX of XX PacBio pipelines

including cases of a single long read (Figure XX).

We found that [Mark add an example here] deemed validated revealed that an isoform for gene XX had an
incorrect alignment.


* % ../bin/paperEvalStats wtc11-support.tsv wtc11-stats.tsv
| name                   | value |
|------------------------+-------|
| genes                  |    88 |
| targets                |    89 |
| transcripts            |    88 |
| novelIsoforms          |    64 |
| maxPipelines           |    33 |
| novelPipelineCnt_ge_1  |    64 |
| novelPipelineCnt_lt_1  |     0 |
| novelPipelineCnt_ge_9  |    39 |
| novelPipelineCnt_lt_9  |    25 |
| novelPipelineCnt_ge_17 |    29 |
| novelPipelineCnt_lt_17 |    35 |
| novelPipelineCnt_ge_25 |    16 |
| novelPipelineCnt_lt_25 |    48 |
| novelPipelineCnt_ge_33 |     2 |
| novelPipelineCnt_lt_33 |    62 |


* % ../bin/paperEvalStats gencode-support.tsv gencode-stats.tsv
| name                       | value |
|----------------------------+-------|
| genes                      |    36 |
| targets                    |    80 |
| transcripts                |    77 |
| gencodeKnown               |    28 |
| gencodeNovel               |    42 |
| gencodeRejected            |     9 |
| gencodeKnowSupportRate     |  82.1 |
| gencodeKnowNotValidated    |  17.9 |
| gencodeNovelSupportRate    |  59.5 |
| gencodeRejectedSupportRate |  22.2 |
| gencodeRejectedSupport     |     2 |

